### PR TITLE
CVE-2011-3190

### PR DESCRIPTION
**Pull request by vasindaniil**
Round 1 Initial submission

### CODE REVIEW

File: cves/CVE-2011-3190.yml [hunk 1]
```diff
@@ -62,7 +67,7 @@ fixes:
    - commit: be3eb28f82250a5c81a1c42216570ebf892aefac
      note: SVN rev 1162960, from the Tomcat website.
    - commit:
-     note:
+     note: SVN rev 1162959, from the Tomcat website.
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: Also revisions 1162959 (Tomcat 6) and 1162958 (Tomcat 7).

File: cves/CVE-2011-3190.yml [hunk 2]
```diff
@@ -62,7 +67,7 @@ fixes:
...
 vccs:
   - commit:
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml:72]: No VCCs?  It appears that this vulnerability has been there since the affected class was first implemented at rev 300781 (see lines 406-416 and 1038-1041 of diff  http://svn.apache.org/viewvc/tomcat/tc5.5.x/trunk/connectors/jk/java/org/apache/coyote/ajp/AjpAprProcessor.java?r1=300781&r2=1162960&pathrev=1162960&diff_format=h).

File: cves/CVE-2011-3190.yml [hunk 3]
```diff
@@ -10,27 +11,27 @@ CWE_instructions: |
...
   set to true.
-curated: false
+curated: true
 
 reported_instructions: |
   Was there a date that this vulnerability was reported to the team? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE data.
   Please enter your date in YYYY-MM-DD format.
-reported:
+reported: 2011-08-19
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: The report date on bugzilla is 08/20/2011 (https://bz.apache.org/bugzilla/show_bug.cgi?id=51698).

File: cves/CVE-2011-3190.yml [hunk 4]
```diff
@@ -10,27 +11,27 @@ CWE_instructions: |
...
 
 announced_instructions: |
   Was there a date that this vulnerability was announced to the world? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE data.
   Please enter your date in YYYY-MM-DD format.
-announced:
+announced: 2011-08-19
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: Same as above.  Since it was disclosed publicly by the reporter w/o any private notifications to the Tomcat team, it was "announced" the same date as reported.

File: cves/CVE-2011-3190.yml [hunk 5]
```diff
@@ -99,7 +104,7 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
+  answer: false
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: The original code was being tested (see http://svn.apache.org/viewvc/tomcat/tc7.0.x/trunk/test/org/apache/coyote/ajp/TestAbstractAjpProcessor.java?view=log&pathrev=1162958), but this case wasn't handled, and none were added after the vulnerability was found.

File: cves/CVE-2011-3190.yml [hunk 6]
```diff
@@ -128,8 +133,8 @@ subsystem:
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
     Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer:
-  name:
+  answer: connector
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml:136]: _coyote_ or _ajp_ may also be appropriate, as the affected classes were part of _org.apache.coyote.ajp_.

File: cves/CVE-2011-3190.yml [hunk 7]
```diff
@@ -171,8 +176,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: yes
+    note: Unexpected input allowed attackers to create a secondary request within the original, due to no input sanitation
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml:180]: It also could be said that this is an issue of complex inputs, as the AJP protocol itself is flexible enough to allow stuffing of requests inside others like this.

File: cves/CVE-2011-3190.yml [hunk 8]
```diff
@@ -202,4 +207,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: The developer did not test the code against edge cases that may put the system into an invalid state
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: So do you think this is a coding mistake or design issue?  It seems more like a design issue to me because the developers didn't think to account for AJP requests that could be purposely malformed.

File: cves/CVE-2011-3190.yml [hunk 9]
```diff
@@ -10,27 +11,27 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: 264
```
Comment (dwelcheck) [inline cves/CVE-2011-3190.yml]: If the issue is really input validation, then is 264 the best fit?  Maybe https://cwe.mitre.org/data/definitions/20.html, https://cwe.mitre.org/data/definitions/88.html, or https://cwe.mitre.org/data/definitions/94.html?

**Comment by dwelcheck** [comment]
+5 upvotes
